packages/typespec-client-generator-core/design-docs/flags.md

```diff
@@ -0,0 +1,35 @@
+# TCGC `tspconfig.yaml` flags
+
+## 1. `generate-protocol-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@protocolAPI` is not set for an operation. Default value is `true`.
+
+## 2. `generate-convenience-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@convenientAPI` is not set for an operation. Default value is `true`.
+
+## 3. `package-name`
+
+The `package-name` override you would like for your generated library. By default individual language emitters should calculate the package name for their library from the `SdkPackage.namespaces` that `tcgc` returns to them.
+
+Currently, emitters don't really use it, but it would be great to move language emitters to use this flag. Java should be excluded from this list because it doesn't apply to their naming structure.
+
+Should remove any logic that derives namespace names from the value of `package-name`.
+
+Default value is `none`.
```

Comment:
Eh, default `none`? Why not `undefined`?

Reply:
Yeah, I messed up on the exact semantics, let me update the docs to `undefined`